### DISCUSSION TITLE
Empty input in line_metric_showcase.rmd

### DISCUSSION THREAD

**Discussion by Ssunoo2** [sailuh/kaiaulu#273] [Q&A] [answered]
https://github.com/sailuh/kaiaulu/blob/9294a9bb92506b76424b0fcc7c07f6b74d5ce53d/vignettes/line_metrics_showcase.Rmd#L93
is giving me an error:

Error in fread(stri_c(stdout, collapse = "\n")) : 
  Input is empty or only contains BOM or terminal control characters

I double checked the file directory pathing and it is correct. SCC is also correct in tools.yml. My understanding is that a .git file is in the project repository. git_repo_path is currently ~/rawdata/git_rep/apr/.git. apr in this case "apr" is the resulting file of "git clone https://github.com/apache/apr" so it is the project files and I can see the .git file with "ls -a" in the terminal. I'm wondering if anyone has had this error and if they have found a fix for it.

**Comment by carlosparadis** [comment]
@Ssunoo2 The path should be under `git_repo`, not `git_rep`. Could you confirm the typo is on the comment and not on the folder structure? 

You can experiment using the tool directly without Kaiaulu as an intermediate to verify if the program can work on input data it expects. To be more precise, the input is the folder. The parameter where you specify .git is a legacy dependency, but you can see the path is formatted inside the function to remove the `.git`: 

https://github.com/sailuh/kaiaulu/blob/9294a9bb92506b76424b0fcc7c07f6b74d5ce53d/R/parser.R#L1257

**Reply by Ssunoo2** [reply]
Confirmed a typo in the comment, not the folder structure. I'm thinking the scc file is not working after all so maybe the filepath is still incorrect. My current filepath is ~/scc-master (the original repo downloaded from the master file). I noticed that the placeholder path in tools.yml was ~/scc/scc but I did not see any second scc file within the first.

**Comment by Ssunoo2** [comment, marked as answer]
The point of confusion was the pathing to the scc binary. The link on the third party tools wiki contains the link to the version that also contains the source code: https://github.com/sailuh/kaiaulu/wiki/Third-Party-Tools-Setup

git cloning the source code results in a file that does not contain the scc executable. Instead the file appropriate for your operating system and unzipping that file will result in the scc binary. Maybe a note specifying not to download the source code would help.
